elixir/lib/cli.ex

```diff
@@ -1,7 +1,9 @@
 defmodule GithubRepoCloner.CLI do
+  alias GithubRepoCloner.Cloner
+  alias GithubRepoCloner.PageIterator
+
   def main(args \\ []) do
-    args
-    |> List.first
-    |> GithubRepoCloner.Cloner.clone
+    username = List.first(args)
+    PageIterator.repeat(&Cloner.clone_page/1, %{username: username, page: 1})
```

Comment:
Had to rethink the cloner logic here. I could have simply tossed the existing logic in a loop that runs until we reach an empty page. That increases the cyclomatic complexity, or in other words the number of scenarios the function handles. I decided it was better to isolate the logic for cloning one page of repos (`Cloner`) and extract the logic for handling pagination elsewhere (`PageIterator`).